traits/editor_factories.py

```diff
@@ -141,6 +150,30 @@ def date_editor():
     return DateEditor
 
 
+def datetime_editor():
+    """ Factory function that returns an editor with date & time for
+    editing DateTime values.
+    """
+    global DateTimeEditor
```

Comment:
I think that this is a premature optimization - if `DateTimeEditor` in TraitsUI grows lots of options, that end up being exposed via metadata then we'll need to take it out; and it may not help much if you always end up creating a custom date editor.  Let's leave it out for now.